modules/tensor_mechanics/src/materials/ComputeThermalExpansionEigenstrainBase.C

```diff
@@ -17,6 +17,10 @@ validParams<ComputeThermalExpansionEigenstrainBase>()
                         "Reference temperature at which there is no "
                         "thermal expansion for thermal eigenstrain "
                         "calculation");
+  params.addCoupledVar("stress_free_aux",
```

Comment:
No. Sorry, I may not have been clear enough. But please do not add all this complexity. Replace `params.addParam<Real>("stress_free_temperature"` with `params.addCoupledVar("stress_free_temperature"`!

MOOSE allows users to still specify a number value, so all old input files would still work. Ditch your helper function and just replace `_stress_free_temperature` with `_stress_free_temperature[_qp]`!

This should be a three line change!